src/groupByKey.ts

```diff
@@ -2,6 +2,15 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((result: GroupsMap<T>, item: T) => {
+    const groupKey = String(item[key]); // Convert the key to a string
```

Comment:
Casting `item[key]` to a string is a safe operation here, but be aware that if the key's value can be an object, the resulting string will be '[object Object]'. If you need to handle non-primitive types, consider implementing a more robust string conversion or using a different method to serialize the key.